lib/posix/clock.c

```diff
@@ -17,6 +18,19 @@
  */
 static struct timespec rt_clock_base;
 
+#ifdef CONFIG_RTC
+static struct device *clock_getrtc(void)
+{
+	static int isfirst = 1;
+	static struct device *rtc_dev;
+
+	if (rtc_dev == NULL) {
+		rtc_dev = device_get_binding(DT_RTC_0_NAME);
```

Comment:
add at least LOG_ERR() when rtc_dev is null

Reply:
I rewrote this